pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -422,3 +441,40 @@ func isBucketNotFound(err interface{}) bool {
 	}
 	return false
 }
+
+// removeNodeServiceAllRule removes a too-permissive capa-upstream ingress rule.
+func removeNodeServiceAllRule(ctx context.Context, session *session.Session, sgID string) error {
+	input := &ec2.RevokeSecurityGroupIngressInput{
+		GroupId: aws.String(sgID),
+		IpPermissions: []*ec2.IpPermission{
+			{
+				IpProtocol: aws.String("tcp"),
+				FromPort:   aws.Int64(30000),
+				ToPort:     aws.Int64(32767),
+				IpRanges: []*ec2.IpRange{
+					{CidrIp: aws.String("0.0.0.0/0")},
+				},
+			},
+			{
+				IpProtocol: aws.String("tcp"),
+				FromPort:   aws.Int64(30000),
+				ToPort:     aws.Int64(32767),
+				Ipv6Ranges: []*ec2.Ipv6Range{
+					{CidrIpv6: aws.String("::/0")},
+				},
+			},
+		},
+	}
+	_, err := ec2.New(session).RevokeSecurityGroupIngressWithContext(ctx, input)
+	if err != nil {
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) && awsErr.Code() == "InvalidPermission.NotFound" {
+			logrus.Debugln("Superfluous Node Port Service ingress rules not found or already deleted")
+			return nil
+		}
+		return err
```

Comment:
I would have imagined, the desired behavior would be to report a warning and continue instead of returning an error here.

Reply:
It's the same behavior as when deleting the ignition bucket and the boostrap ssh rule. If we change, we'll need to make it consistent with the others.